lib/nats.js

```diff
@@ -843,6 +865,9 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
  */
 
 Client.prototype.subscribe = function(subject, opts, callback) {
+  if (this.closed) {
+    this.emit('error', SUB_CONN_CLOSED_ERR);
```

Comment:
If the connection is dropped and we are trying to reconnect these will all work with no errors. If you closed the connection explicitly and then make a call to publish, subscribe, flush etc. this will get triggered.